src/app/Dashboard/DashboardLayoutCreateModal.tsx

```diff
@@ -76,114 +98,148 @@ export const DashboardLayoutCreateModal: React.FC<DashboardLayoutCreateModalProp
-        <TextInput
+      <FormSection>
+        {isCreateModal && (
+          <FormGroup label={'Template'} fieldId="template" isRequired height="35em" validated={nameValidated}>
+            <div style={{ border: '1px solid var(--pf-global--BorderColor--100)', height: '33em' }}>
+              <LayoutTemplatePicker onTemplateSelect={onTemplateSelect} />
+            </div>
+          </FormGroup>
+        )}
+        <FormGroup
+          label={t('DashboardLayoutCreateModal.NAME.LABEL')}
+          fieldId="name"
+          helperText={t('DashboardLayoutCreateModal.NAME.HELPER_TEXT')}
+          helperTextInvalid={errorMessage}
           isRequired
-          type="text"
-          id="name"
-          name="name"
-          aria-describedby={'name-helper'}
-          value={name}
-          onChange={handleNameChange}
-          onKeyUp={handleKeyUp}
-          autoFocus={true}
-          autoComplete="on"
-        />
-      </FormGroup>
+          validated={nameValidated}
+        >
+          <TextInput
+            isRequired
+            type="text"
+            id="name"
+            name="name"
+            aria-describedby={'name-helper'}
+            value={name}
+            onChange={handleNameChange}
+            onKeyUp={handleKeyUp}
+            autoFocus={true}
+            autoComplete="on"
+            validated={nameValidated}
+          />
+        </FormGroup>
+      </FormSection>
     );
-  }, [t, validated, errorMessage, name, handleNameChange, handleKeyUp]);
+  }, [t, isCreateModal, nameValidated, errorMessage, name, onTemplateSelect, handleNameChange, handleKeyUp]);
 
   const actionGroup = React.useMemo(() => {
     return (
       <ActionGroup>
-        <Button variant="primary" onClick={handleSubmit} isAriaDisabled={validated !== 'success'}>
-          {t('CREATE', { ns: 'common' })}
+        <Button variant="primary" onClick={handleSubmit} isAriaDisabled={nameValidated !== 'success'}>
+          {isCreateModal ? t('CREATE', { ns: 'common' }) : t('RENAME', { ns: 'common' })}
         </Button>
         <Button variant="link" onClick={handleClose}>
           {t('CANCEL', { ns: 'common' })}
         </Button>
       </ActionGroup>
     );
-  }, [t, handleSubmit, validated, handleClose]);
+  }, [t, handleSubmit, handleClose, nameValidated, isCreateModal]);
+
+  const header = (
+    <React.Fragment>
+      <Split hasGutter>
+        <SplitItem isFilled>
+          <Title id="modal-custom-header-label" headingLevel="h1" size={TitleSizes['2xl']}>
+            {isCreateModal
+              ? t('DashboardLayoutCreateModal.CREATE_LAYOUT')
+              : t('DashboardLayoutCreateModal.RENAME_LAYOUT')}
+          </Title>
+        </SplitItem>
+      </Split>
+    </React.Fragment>
+  );
 
   return (
     <Modal
+      aria-label="Create a new dashboard layout" // FIXME: i18n
+      width={isCreateModal ? '110em' : '40%'}
       appendTo={portalRoot}
       isOpen={props.visible}
       variant={ModalVariant.large}
```

Comment:
Remove variant and use set `width` and `height` (use css class)  with percentage would be better? This way, it always resize based on viewport on both directions?